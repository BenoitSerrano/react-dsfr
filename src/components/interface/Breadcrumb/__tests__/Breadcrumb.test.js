import { render, screen } from '@testing-library/react';
import { Breadcrumb, BreadcrumbButton, BreadcrumbItem } from '..';

describe('<Breadcrumb />', () => {
  it('should render breadcrumb properly', () => {
    render(
      <Breadcrumb label="fil d'ariane">
        <BreadcrumbButton>
          Voir le fil d&paos;ariane
        </BreadcrumbButton>
        <BreadcrumbItem href="#">Home</BreadcrumbItem>
        <BreadcrumbItem href="#">Productions</BreadcrumbItem>
        <BreadcrumbItem current>Production 1234567899</BreadcrumbItem>
      </Breadcrumb>,
    );
    const nav = screen.getByTestId('bc-nav');
    expect(nav.className).toBe('rf-breadcrumb');
    const bcList = screen.getByTestId('bc-list');
    expect(bcList.className).toBe('rf-breadcrumb__list');
    const button = screen.getByTestId('bc-button');
    expect(button.className).toBe('rf-breadcrumb__button');
    const items = screen.getAllByTestId('bc-item');
    expect(items).toHaveLength(3);
    expect(items[0].className).toBe('rf-breadcrumb__item');
    expect(items[2].className).toMatch(/rf-breadcrumb__item--current/);
    expect(nav).toMatchSnapshot();
  });
});
