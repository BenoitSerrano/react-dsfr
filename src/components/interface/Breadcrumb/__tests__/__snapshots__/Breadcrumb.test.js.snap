// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render breadcrumb properly 1`] = `
<nav
  aria-label="fil d'ariane"
  class="rf-breadcrumb"
  data-testid="bc-nav"
>
  <button
    class="rf-breadcrumb__button"
    data-testid="bc-button"
    hidden=""
    type="button"
  >
    Voir le fil d&paos;ariane
  </button>
  <ol
    class="rf-breadcrumb__list"
    data-testid="bc-list"
  >
    <li
      aria-current="false"
      class="rf-breadcrumb__item"
      data-testid="bc-item"
    >
      <a
        href="#"
      >
        Home
      </a>
    </li>
    <li
      aria-current="false"
      class="rf-breadcrumb__item"
      data-testid="bc-item"
    >
      <a
        href="#"
      >
        Productions
      </a>
    </li>
    <li
      aria-current="page"
      class="rf-breadcrumb__item rf-breadcrumb__item--current"
      data-testid="bc-item"
    >
      Production 1234567899
    </li>
  </ol>
</nav>
`;
